### PR TITLE
HEEDLS-623 Add search function to delegate groups

### PR DESCRIPTION
**Pull request by wilkinsm**
### JIRA link
[https://softwiretech.atlassian.net/browse/HEEDLS-623](https://softwiretech.atlassian.net/browse/HEEDLS-623)

### Description
Add search by group label field to delegate groups page.
(Includes a non-Javascript version)

### Screenshots
**Desktop**
image

**Desktop - No JS version**
image

**Tablet**
image

**Mobile**
image

-----
### Developer checks
(Leave tasks unticked if they haven't been appropriate for your ticket.)

I have:
- [x] Run the formatter and made sure there are no IDE errors.
- [x] Written tests for the changes (accessibility tests, unit tests for controller, data services, services, view models, etc)
- [x] Manually tested my work with and without JavaScript. Full manual testing guidelines can be found here: https://softwiretech.atlassian.net/wiki/spaces/HEE/pages/6703648740/Testing
- [ ] Updated/added documentation in Swiki and/or Readme. Links (if any) are below:
  - [doc_1_here](link_1_here)
- [ ] Updated my Jira ticket with information about other parts of the system that were touched as part of the MR and have to be sanity tested to ensure nothing’s broken.
- [x] Scanned over my own MR to ensure everything is as expected.

### CODE REVIEW

**Review by JBsoftwire — APPROVED** [review]
Looks good.

File: DigitalLearningSolutions.Web.Tests/ViewModels/TrackingSystem/Delegates/DelegateGroups/DelegateGroupsViewModelTests.cs [hunk 1]
```diff
@@ -110,6 +113,7 @@ public void DelegateGroupsViewModel_filters_should_be_set()
             // Then
             model.Filters.Should().BeEquivalentTo(expectedFilters);
             model.DelegateGroups.First().Name.Should().BeEquivalentTo("K");
+            model.SearchString.Should().Be("K");
```
Comment (AlexJacksonDS) [inline DigitalLearningSolutions.Web.Tests/ViewModels/TrackingSystem/Delegates/DelegateGroups/DelegateGroupsViewModelTests.cs:116]: Probably need to rename this test if we're asserting this as well now (or make a separate test for it)

**Review by AlexJacksonDS — CHANGES_REQUESTED** [review]
Just one minor naming issue in the unit tests.

File: DigitalLearningSolutions.Web.Tests/ViewModels/TrackingSystem/Delegates/DelegateGroups/DelegateGroupsViewModelTests.cs (hunk 1 above)
Comment (wilkinsm) [reply DigitalLearningSolutions.Web.Tests/ViewModels/TrackingSystem/Delegates/DelegateGroups/DelegateGroupsViewModelTests.cs:116]: Yeah no worries. Done.

**Review by AlexJacksonDS — APPROVED** [review]

**Pull request merged**